src/CentralDesktop/Stomp/Connection.php

```diff
@@ -139,9 +139,14 @@ function _makeConnection() {
             $host = $url['host'];
 
             $this->connectedHost = $host;
-            $this->_socket       = @fsockopen($brokerUri, $port, $connect_errno, $connect_errstr, $this->_connect_timeout_seconds);
+
+            $hostname = $host;
+            if (array_key_exists('scheme', $url)) {
+                $hostname = "{$url['scheme']}://$host";
+            }
+            $this->_socket = @fsockopen($hostname, $port, $connect_errno, $connect_errstr, $this->_connect_timeout_seconds);
```

Comment:
I was going to say you shouldn't pass "tcp://hostname" into the hostname param, but the docs say I'm wrong.